### PR TITLE
Fix issues when push entity whose key is long type

### PR DESCRIPTION
**Pull request by chinadragon0515**
### Issues
*This pull request fixes issue #380 .*  

### Description
Make put can work with key with type long(int64)

### Checklist (Uncheck if it is not completed)
- [ x ] Test cases added
- [ x ] Build and test with one-click build and test script passed

### CODE REVIEW

File: src/Microsoft.Restier.EntityFramework/Submit/ChangeSetPreparer.cs [hunk 1]
```diff
@@ -230,6 +230,12 @@ private static object ConvertToEfValue(Type type, object value)
                 return (TimeSpan)timeOfDayValue;
             }
 
+            // In case key is long type, when put an entity, key value will be from key parsing which is type of int
+            if (value is int && type == typeof(long))
```
Comment (rayao) [inline src/Microsoft.Restier.EntityFramework/Submit/ChangeSetPreparer.cs]: Could type be `Nullable<long>`?

Comment (chinadragon0515) [inline src/Microsoft.Restier.EntityFramework/Submit/ChangeSetPreparer.cs]: No, as this is only used for key which is parsed by RESTier, and for other update properties, they are parsed by WebAPi, and type is always correct which I have verified.

File: test/ODataEndToEndTests/Microsoft.Restier.WebApi.Test.Services.Trippin/Models/Person.cs [hunk 2]
```diff
@@ -26,7 +26,7 @@ public class Person
         [MaxLength(26), MinLength(1)]
         public string LastName { get; set; }
 
-        public int? Age { get; set; }
+        public long? Age { get; set; }
```
Comment (lewischeng-ms) [inline test/ODataEndToEndTests/Microsoft.Restier.WebApi.Test.Services.Trippin/Models/Person.cs]: `Age` is not key property?

Comment (lewischeng-ms) [inline test/ODataEndToEndTests/Microsoft.Restier.WebApi.Test.Services.Trippin/Models/Person.cs]: If it is to test the logic modified in `ChangeSetPreparer`, please make sure there IS some test data whose `Age` values are null.

**Comment by lewischeng-ms** [comment]
Looks good to me for the other part.

File: test/ODataEndToEndTests/Microsoft.Restier.WebApi.Test.Services.Trippin/Models/Person.cs (hunk 2 above)
Comment (chinadragon0515) [inline test/ODataEndToEndTests/Microsoft.Restier.WebApi.Test.Services.Trippin/Models/Person.cs]: Yes, I make some test specified null for Age now.

**Pull request merged**